docs/basic/sort-intro.md

```diff
@@ -1,23 +1,49 @@
-**There are many sorting algorithms**, and most of them are different in nature.
+Last translate with upstream: [08eb0c5](https://github.com/OI-wiki/OI-wiki/commit/08eb0c53b1ad55b518b0f5f7a569c25b8c1215b2#diff-fe072ff43c0ab8facb3edd321f4ac26c2d1239221c37b5bc93fa00eddbd69c51)
 
-## Stability
+This article will briefly introduce sorting algorithms.
+
+## Introduction
+
+In computer science, a **sorting algorithm** is an algorithm that puts a specific group of data into some order. There are many sorting algorithms, and most of them have different natures.
+
+## Properties
+
+### Stability
 
 Stability refers to whether the relative order of equal elements has changed after sorting.
 
+Sorting algorithm with stability will preserve the relative order of equal elements. That is, when we call that a sorting algorithm is stable, we mean that if there are two records $R$ and $S$ compared as equal, and in the original list $R$ comes before $S#, then in sorted list $R$ will also comes before $S$.
+
 Radix sort, counting sort, insertion sort, bubble sort, and merging sort are stable sorting algorithms.
 
 Selection sort, heap sort, and quick sort are not stable sorting algorithms.
 
-## Time complexity
+### Time complexity
+
+Main article: [Complexity](./complexity.md)
+
+Time complexity is used to measure the relationship between the running time of an algorithm and the input scale. Commonly it is represented as $O()$
 
-Time complexity is used to measure the relationship between the running time of an algorithm and the input scale. Similarly, there is space complexity, which is used to describe the scale of the memory consumption of the algorithm.
+<!---Similarly, there is space complexity, which is used to describe the scale of the memory consumption of the algorithm.-->
 
-The simple method of calculating the complexity is generally to count the number of executions of "simple operations", and sometimes it can be approximated by directly counting the number of cycles.
+The simple method of calculating the complexity is generally to count the number of executions of elementary operations, and sometimes it can be approximated by directly measuring the number of loops.
 
-Time complexity is divided into worst time complexity, average time complexity, best time complexity and so on. The worst time complexity usually needs to be considered in the OI competitions, because it represents the lower bound of the algorithm's operation, and there will be no worse results in the evaluation.
+Time complexity is divided into worst-case complexity, average-case complexity, best-case complexity and so on. The worst-case complexity usually needs to be considered in competitive programming, because it represents the lower bound of the algorithm's operation, and there will be no worse results in the evaluation.
```

Comment:
```suggestion
Time complexity is divided into worst-case complexity, average-case complexity, best-case complexity and so on. The worst-case complexity usually needs to be considered in competitive programming, because it represents the algorithm's worst performance, and no input will bring performance worse than that.
```

Reply:
emmmm 下界不直译喵

Reply:
感觉是没写出来执行用时那个味道 🤔